ruby/lib/google/protobuf.rb

```diff
@@ -31,6 +31,15 @@
 # require mixins before we hook them into the java & c code
 require 'google/protobuf/message_exts'
 
+# We define these before requiring the platform-specific modules.
+# That way the module init can grab references to these.
+module Google
+  module Protobuf
+    class Error < RuntimeError; end
```

Comment:
Prefer StandardError for this use case. RuntimeError is conventionally used for invalid operations (i.e. attempting to execute invalid code), rather than errors caused by the environment.